measured.go

```diff
@@ -124,14 +127,12 @@ func (c *conn) Read(b []byte) (int, error) {
 	return n, err
 }
 
-func (c *conn) Close() error {
-	if atomic.CompareAndSwapInt32(&c.closed, 0, 1) {
-		err := c.Conn.Close()
-		// Wait for tracking to finish
-		<-c.trackingFinished
-		return err
-	}
```

Comment:
I'm only noting this because we're investigating possible resource leaks related to these `measured.Conn` objects: I don't _think_ this closing sequence could result in a leak.  It looks like this `if` clause will only evaluate once and the tracking goroutine will always end up checking the value of `c.closed`, then signaling on `c.trackingFinished`.

However, I totally agree that the existing logic is complicated and that your new closing sequence is easier to follow and reason about.  So I think this is a positive change!